packages/plugins/response-cache/src/redis-cache.ts

```diff
@@ -0,0 +1,115 @@
+  buildRedisResponseOpsKey?: BuildRedisResponseOpsKey;
+};
+
+export const createRedisCache = (params: RedisCacheParameter): Cache => {
+  const store = params.redis;
+
+  const buildRedisEntityId = params?.buildRedisEntityId ?? defaultBuildRedisEntityId;
+  const buildRedisResponseOpsKey = params?.buildRedisResponseOpsKey ?? defaultBuildRedisResponseOpsKey;
+
+  function purgeEntity(entity: string) {
+    // find the responseIds for the entity
+    store.smembers(entity).then(function (responseIds) {
+      // and purge each response since they contained the entity data
+      store.del(
+        responseIds.map(responseId => {
+          return buildRedisResponseOpsKey(responseId);
+        })
+      );
+      store.del(responseIds);
+    });
+
+    // if purging an entity like Comment, then also purge Comment:1, Comment:2, etc
+    if (!entity.includes(':')) {
+      store.keys(`${entity}:*`).then(function (entityKeys) {
+        for (const entityKey of entityKeys) {
+          // and purge any reponses in each of those entity keys
+          store.smembers(entityKey).then(function (responseIds) {
+            // if purging an entity check for associated operations containing that entity
+            store.del(
+              responseIds.map(responseId => {
+                return buildRedisResponseOpsKey(responseId);
+              })
+            );
+            // and purge each response since they contained the entity data
+            store.del(responseIds);
+          });
+        }
+
+        // then purge the entityKeys like Comment:1, Comment:2 etc
+        store.del(entityKeys);
+      });
+    }
+
+    // and then purge the entity itself
+    store.del(entity);
+  }
+
+  return {
+    set(responseId, result, collectedEntities, ttl) {
+      if (ttl === Infinity) {
+        store.set(responseId, JSON.stringify(result));
+      } else {
+        // set the ttl in milliseconds
+        store.set(responseId, JSON.stringify(result), 'PX', ttl);
+      }
+
+      const responseKey = buildRedisResponseOpsKey(responseId);
+
+      for (const { typename, id } of collectedEntities) {
+        // Adds a key for the typename => response
+        store.sadd(typename, responseId);
+        // Adds a key for the operation => typename
+        store.sadd(responseKey, typename);
+
+        if (id) {
+          const entityId = buildRedisEntityId(typename, id);
+          // Adds a key for the typename:id => response
+          store.sadd(entityId, responseId);
+          // Adds a key for the operation => typename:id
+          store.sadd(responseKey, entityId);
+        }
+      }
+    },
+    get(responseId) {
+      return (
+        store.get(responseId).then(function (result) {
+          return result && JSON.parse(result);
+        }) ?? null
+      );
+    },
+    invalidate(entitiesToRemove) {
+      for (const { typename, id } of entitiesToRemove) {
+        purgeEntity(id != null ? buildRedisEntityId(typename, id) : typename);
+      }
```

Comment:
```suggestion
      const keys$: Promise<string[]>[] = [];
      for (const { typename, id } of entitiesToRemove) {
        keysPromises.push(purgeEntity(id != null ? buildRedisEntityId(typename, id) : typename));
      }
      
      Promise.all(keys$).then(keys => store.del(keys.flat()));
```

Reply:
Resolves promises. Though instead of ` Promise<string[]>[]`  the purgeEntity returns  Promise<string[]>